src/services/accommodation.service.js

```diff
@@ -16,10 +19,19 @@ const AccommodationService = {
     return accommodations;
   },
 
-  async getAllAccommodationsByAvailability(isAvailable) {
+  async getAllAccommodationsByFilter(req) {
+    const { isAvailable, address, name, costLessOr, costGreaterOr, services } = req.query;
+    const condition = {};
+    if (isAvailable) condition.isAvailable = isAvailable;
+    if (address) condition.address = { [Op.startsWith]: address };
+    if (services) condition.services = { [Op.startsWith]: services };
+    if (name) condition.name = { [Op.startsWith]: name };
```

Comment:
Instead of using `startsWith`, consider using sql case-insensitive comparison operator with multiple wildcards on both sides
```
[Op.iLike]: `%${address}%'
```

Reply:
Thank you. fixed